### PR TITLE
Add functions to support refresh tokens

### PR DESCRIPTION
**Pull request by turt2live**
MSC: https://github.com/matrix-org/matrix-doc/pull/2918 (stable, but not formalized)
For https://github.com/vector-im/element-web/issues/18698
For https://github.com/matrix-org/matrix-react-sdk/pull/7802

---
Here's what your changelog entry will look like:

## ✨ Features
 * Add functions to support refresh tokens ([\#2178](https://github.com/matrix-org/matrix-js-sdk/pull/2178)).

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -6772,6 +6775,29 @@ export class MatrixClient extends EventEmitter {
...
+    public refreshToken(refreshToken: string): Promise<IRefreshTokenResponse> {
+        return this.http.authedRequest(
+            undefined, Method.Post, "/_matrix/client/v1/refresh",
+            undefined, {
+                refresh_token: refreshToken,
+            }, {
+                prefix: "", // set manually
+                inhibitLogoutEmit: true, // we don't want to cause logout loops
+            },
+        );
```
Comment (t3chguy) [inline src/client.ts:6798]: can we be consistent with how many args we do per line?

File: src/client.ts [hunk 2]
```diff
@@ -6772,6 +6775,29 @@ export class MatrixClient extends EventEmitter {
...
+     * @return {Promise<IRefreshTokenResponse>} Resolves to the new token.
+     * @return {module:http-api.MatrixError} Rejects with an error response.
+     */
+    public refreshToken(refreshToken: string): Promise<IRefreshTokenResponse> {
+        return this.http.authedRequest(
+            undefined, Method.Post, "/_matrix/client/v1/refresh",
+            undefined, {
+                refresh_token: refreshToken,
+            }, {
+                prefix: "", // set manually
```
Comment (t3chguy) [inline src/client.ts:6795]: This should be using `PREFIX_V1` and `/refresh` as the path

**Review by t3chguy — APPROVED** [review]
LGTM otherwise

File: src/client.ts (hunk 1 above)
Comment (turt2live) [reply src/client.ts:6798]: can I say no? :P 

(I don't know what I was doing here tbh)

**Pull request merged**